ebms-provider/src/main/kotlin/no/nav/emottak/ebms/model/EbmsMessageExt.kt

```diff
@@ -1,29 +1,44 @@
 package no.nav.emottak.ebms.model
 
 import no.nav.emottak.ebms.persistence.repository.EbmsMessageDetailsRepository
+import no.nav.emottak.ebms.persistence.repository.EventsRepository
 import no.nav.emottak.ebms.util.marker
 import no.nav.emottak.ebms.validation.SignaturValidator
 import no.nav.emottak.message.model.EbmsMessage
+import no.nav.emottak.message.model.Event
 import no.nav.emottak.message.model.PayloadMessage
 import no.nav.emottak.message.model.SignatureDetails
 import no.nav.emottak.message.model.log
 import no.nav.emottak.message.model.toEbmsMessageDetails
 import java.sql.SQLException
 import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
 
 fun EbmsMessage.sjekkSignature(signatureDetails: SignatureDetails) {
     SignaturValidator.validate(signatureDetails, this.dokument!!, if (this is PayloadMessage) listOf(this.payload) else listOf())
     log.info("Signatur OK")
 }
 
+@OptIn(ExperimentalUuidApi::class)
+fun EbmsMessage.saveEvent(message: String, eventsRepository: EventsRepository) {
+    log.info(this.marker(), message)
+    eventsRepository.updateOrInsert(
+        Event(
+            eventId = Uuid.random(),
+            referenceId = Uuid.parse(this.requestId),
+            messageId = this.messageId,
+            eventMessage = message
+        )
+    )
+}
```

Comment:
Jeg funderte litt: Hva var årsak til at vi valgte å ta i bruk kotlin sin UUID i stedet for den innebygde? Sikkert allerede vært diskutert på et delingsmøte eller noe.. :sweat_smile: